code/game/objects/machinery/vending/vending.dm

```diff
@@ -23,64 +51,124 @@
+	var/isshared = FALSE
 
-	var/product_slogans = "" //String of slogans separated by semicolons, optional
-	var/product_ads = "" //String of small ad messages in the vending screen - random chance
+	/*These are lists that are made null after they're used, their use is solely to fill the inventory of the vendor on Init.
+	They use the following pattern in case if it doenst pertain to a tab:
+	list(
+		/type/path = amount,
+		/type/path2 = amount2,
+	)
+	if you want the item to pertain to a tab you use:
+	list(
+		"Tab1" = list(
+			/type/path = amount.
+			/type/path2 = amount2,
+		),
+		"Tab2" = list(
+			/type/path3 = amount3,
+			/type/path4 = amount4,
+		)
+	)
+	*/
+	/// Normal products that are always available on the vendor.
+	var/list/products = list()
+	/// Contraband products that are only available on vendor when hacked.
+	var/list/contraband = list()
+	/// Premium products that are only available when using a coin to pay for it.
+	var/list/premium = list()
+	/// Prices for each item, list(/type/path = price), items not in the list don't have a price.
+	var/list/prices = list()
+
+	/// String of slogans separated by semicolons, optional
+	var/product_slogans = ""
+	///String of small ad messages in the vending screen - random chance
+	var/product_ads = ""
+
+	//These are where the vendor holds their item info with /datum/vending_product
+
+	///list of /datum/vedning_product's that are always available on the vendor
 	var/list/product_records = list()
+	///list of /datum/vedning_product's that are available when vendor is hacked.
 	var/list/hidden_records = list()
+	///list of /datum/vedning_product's that are available on the vendor when a coin is used.
 	var/list/coin_records = list()
+
 	var/list/slogan_list = list()
-	var/list/small_ads = list() // small ad messages in the vending screen - random chance of popping up whenever you open it
-	var/vend_reply //Thank you for shopping!
+	/// small ad messages in the vending screen - random chance of popping up whenever you open it
+	var/list/small_ads = list()
+	///Message spoken by the vending machine when a item is vended
+	var/vend_reply
+	///When was last time we spoke when vending
 	var/last_reply = 0
-	var/last_slogan = 0 //When did we last pitch?
-	var/slogan_delay = 600 //How long until we can pitch again?
-	var/icon_vend //Icon_state when vending!
-	var/icon_deny //Icon_state when vending!
-	var/seconds_electrified = 0 //Shock customers like an airlock.
-	var/shoot_inventory = FALSE //Fire items at customers! We're broken!
-	var/shut_up = FALSE //Stop spouting those godawful pitches!
-	var/extended_inventory = FALSE //can we access the hidden inventory?
+	///Last time we spoke our slogan
+	var/last_slogan = 0
+	//The interval between slogans.
+	var/slogan_delay = 600
+	///Icon state when successfuly vending
+	var/icon_vend
+	///Icon state when failing to vend, be it by no access or money.
+	var/icon_deny
+	///how many time we have left electrified
+	var/seconds_electrified = 0
```

Comment:
Ho much time?
what measure is this byond ticks or what

Reply:
need to check

Reply:
![image](https://user-images.githubusercontent.com/29824735/115599252-c43a5380-a2b1-11eb-8374-6d08f69a4708.png)